### ISSUE TITLE
Add Power Ups

### ISSUE THREAD

**Issue by kainthcool** [vineetjc/pygame-Coin-Fall-#19]
we can add dropping power-ups which may slow the pace of the game for some time. Or some objects which can fasten up the pace.

**Comment by vineetjc** [comment]
Awesome idea! Let's see how it goes, if it's good, we'll make a game mode out of it. Would you like to take this up?

**Comment by kainthcool** [comment]
Yeah sure.

**Comment by vineetjc** [comment]
@kainthcool I'll assign it to you, see if it's easier making a separate branch or if you can do all the changes to make it as a game mode

**Comment by vineetjc** [comment]
@kainthcool please join the Gitter channel

**Comment by vineetjc** [comment]
From #52 quoting @amrit-choudhary 

Buy upgrades to enhance your cart. Upgrades can include:

    Better speed
    Better size
    Magnet effect (coins are pulled towards the cart)
    Health (will take more bomb hits to die)

**Comment by amrit-choudhary** [comment]
Upgrade the cart is a permanent thing.
Powerups are temporary, which are active for a short while during gameplay.

**Comment by vineetjc** [comment]
@amrit-choudhary don't you think that would just make it 'unfair' later on? In the sense, each time one plays and with a possible goal of scoring as high as possible, then what you're suggesting destroys the incentive of scoring high 'fairly'. Instead of a permanent change, let it be a power-up. What do you think?

**Comment by amrit-choudhary** [comment]
Powerups and upgrades, both can be a thing. Just see any mobile game. Both serve a different purpose. Powerups are more fun and action oriented. Upgrades have the benefit of giving you some progress every time.

**Comment by vineetjc** [comment]
@amrit-choudhary I suppose another game mode with this 'upgrade' progress could do instead of having it in say 'Classic' mode?

**Comment by amrit-choudhary** [comment]
@vineetjc Yeah, that's possible. Like a story mode, where you progress and upgrade. And a arcade mode, where on every run you start from 0 and try to get as high as possible. 
While in story mode, your consecutive highscores will be higher, because you have upgrades.

**Comment by vineetjc** [comment]
@amrit-choudhary I think we can do this, and cap the max upgrade?

**Comment by amrit-choudhary** [comment]
Yeah. In this example, the upgrades increase powerup's power.
![image](https://user-images.githubusercontent.com/6888341/50180294-3a6bcb80-032f-11e9-9a7a-1fecdbd8e6e6.png)
